tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -13188,11 +13158,11 @@ void ExecutionTest::WaveSizeTest() {
       "\r\n"
       "// Note: WAVESIZE will be defined via compiler option -D\r\n"
       "[wavesize(WAVESIZE)]\r\n"
-      "[numthreads(" strinfigy(
-          MAX_WAVESIZE) "*2,1,1)]\r\n"
-                        "void main(uint3 tid : SV_DispatchThreadID ) { \r\n"
-                        "  data[tid.x].count = WaveActiveSum(1); \r\n"
-                        "}\r\n";
+      "[numthreads(1,1,1)]\r\n"
```

Comment:
I suppose with the different approach of using `WaveGetLaneCount()`, a group size of 1 is _technically_ sufficient to output the value from the first lane, however, it is a degenerate case.  While this hopefully won't be handled any differently than other dimensions, it's pretty likely that group sizes do impact the picking of wave sizes for drivers.  One could argue that you shouldn't really request a wave size larger than your thread group size anyway, since how would you possibly use that larger wave size?

I think the point of my rambling is that it might be better to test with a threadgroup size that's a multiple of the largest possible wave size of 128.

Reply:
And if doing so, you'll want to use comparison with the SV_DispatchThreadID to only write a result out on ID (0,0,0).